File: Ejercicios Propuestos 2/Ejercicio 16/Ejercicio 16.jl
# Bucle de validación 
while true
    println("Ingrese un número natural:")
    N = parse(Int, readline())
    
    # Si N < 1 Entonces se activa el error
    if N < 1
        println("¡Error! Debe ingresar un número natural (mayor o igual a 1).")
        println("Por favor, intente de nuevo.")
        println("--------------------------------------------------")
    else
        global N
        break # Hasta Que N >= 1
    end
end

digitos = 0
temp = N # Guardamos N en una variable temporal para no borrar el original

# Bucle Mientras
while temp > 0
    global temp, digitos
    # Dividimos entre 10; la función div corta directamente los decimales
    temp = div(temp, 10) 
    
    # Sumamos 1 al contador por cada corte que hacemos
    digitos = digitos + 1   
end

println("El número ", N, " tiene ", digitos, " dígito(s).")

File: Ejercicios Propuestos 2/Ejercicio 16/Ejercicio 16.java
import java.util.Scanner;

public class EJERCICIO16DeGRUPO2 {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);

        // Definir N, temp, digitos Como Entero
        int N, temp, digitos;

        // Validación de entrada (Repetir... Hasta Que N >= 1)
        do {
            System.out.println("Ingrese un número natural:");
            N = sc.nextInt();

            if (N < 1) {
                System.out.println("¡Error! Debe ingresar un número natural (mayor o igual a 1).");
                System.out.println("Por favor, intente de nuevo.");
                System.out.println("--------------------------------------------------");
            }
        } while (N < 1); // Se repite MIENTRAS N sea menor que 1

        digitos = 0;
        temp = N; // Guardamos N en una variable temporal para no borrar el original

        // Mientras temp > 0 Hacer
        while (temp > 0) {
            // Dividimos entre 10; en Java la división de enteros se trunca automáticamente
            temp = temp / 10;

            // Sumamos 1 al contador por cada corte que hacemos
            digitos = digitos + 1;
        }

        // Mostrar el resultado
        System.out.println("El número " + N + " tiene " + digitos + " dígito(s).");

        sc.close();
    }
}
